pkg/util/names/names.go

```diff
@@ -95,7 +164,7 @@ func GenerateWorkName(kind, name, namespace string) string {
 	}
 	hash := fnv.New32a()
 	hashutil.DeepHashObject(hash, workName)
-	return fmt.Sprintf("%s-%s", name, rand.SafeEncodeString(fmt.Sprint(hash.Sum32())))
+	return fmt.Sprintf("%s-%s", ShortNameLabelValue(name), rand.SafeEncodeString(fmt.Sprint(hash.Sum32())))
```

Comment:
Does the compatibility problem exist?  After upgrading Karmada, everything works right? like status syncing, and work syncing.

Also, Even if we truncate the name, the problem still exists, like the name is len(abcabacacb...abc)=63, so I prefer to move it to annotations.

Reply:
`compatibility problem` do exist. `work` with name reached limit is very few, I use a script to delete such one or two work cr.
Then regenerated `work` works right.

Annotations is also a good place. I don't object.

**BUT**, **However** it happend in my **PRODUCTION**, I can't wait. With this pr, my production fixed.

Feel free to close this pr, if finally deside to use annotations.

Reply:
I will add it to my queue and will try to review it this week(or next week, too much on my plate recently.).